### PR TITLE
Fb res 206 investigate convergence oilphase

### PR DESCRIPTION
**Pull request by alexandrebbruno**

### CODE REVIEW

File: Reaktoro/Thermodynamics/EOS/CuibicEOS-test.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    Pressure = 100000.0
+    b = [3.0,
+         122.01687012,
+         1.0,
+         63.50843506,
+         0.0]
+   
+    solver.approximate(state, Temperature, Pressure, b)
+    
+    state.properties()
```
Comment (armandoeng) [inline Reaktoro/Thermodynamics/EOS/CuibicEOS-test.py]: Should we use regression here?

File: Reaktoro/Thermodynamics/EOS/CubicEOS.cpp [hunk 2]
```diff
@@ -322,13 +322,24 @@ struct CubicEOS::Impl
         const ChemicalScalar BT = 2*(epsilon*sigma - epsilon - sigma)*beta*betaT + qT*beta - (epsilon + sigma - q)*betaT;
         const ChemicalScalar CT = -3*epsilon*sigma*beta*beta*betaT - qT*beta*beta - 2*(epsilon*sigma + q)*beta*betaT;
 
-        // Calculate cubicEOS roots using cardano's method
-        auto cubicEOS_roots = realRoots(cardano(1, A.val, B.val, C.val));
+        // Calculate cubicEOS roots
+        auto roots = cardano(1, A.val, B.val, C.val);
```
Comment (armandoeng) [inline Reaktoro/Thermodynamics/EOS/CubicEOS.cpp]: if root doesn't change, put a const on it.

File: Reaktoro/Thermodynamics/EOS/CubicEOS.cpp [hunk 3]
```diff
@@ -322,13 +322,24 @@ struct CubicEOS::Impl
...
+
+        //verify if all cubicEOS_roots are greater than bmix
+        std::vector<double> cubicEOS_roots;
+        cubicEOS_roots.reserve(3);
+        if (std::get<0>(roots).imag() == 0 && std::get<0>(roots).real() >= bmix.val)
```
Comment (armandoeng) [inline Reaktoro/Thermodynamics/EOS/CubicEOS.cpp]: I think spaces between each `if`would help to clear that are 3 independent `ifs`

File: Reaktoro/Thermodynamics/EOS/CubicEOS.cpp [hunk 4]
```diff
@@ -322,13 +322,24 @@ struct CubicEOS::Impl
...
+            cubicEOS_roots.push_back(std::get<0>(roots).real());
+        if (std::get<1>(roots).imag() == 0 && std::get<1>(roots).real() >= bmix.val)
+            cubicEOS_roots.push_back(std::get<1>(roots).real());
+        if (std::get<2>(roots).imag() == 0 && std::get<2>(roots).real() >= bmix.val)
+            cubicEOS_roots.push_back(std::get<2>(roots).real());
 
         // All possible Compressibility factor
         std::vector<ChemicalScalar> Zs;
-        if (cubicEOS_roots.size() == 1)
+        if (cubicEOS_roots.size() == 1 || cubicEOS_roots.size() == 2)
```
Comment (armandoeng) [inline Reaktoro/Thermodynamics/EOS/CubicEOS.cpp]: Store size:
```c++
const auto cubic_size = cubicEOS_roots.size();
```

File: Reaktoro/Thermodynamics/EOS/CuibicEOS-test.py (hunk 1 above)
Comment (alexandrebbruno) [reply Reaktoro/Thermodynamics/EOS/CuibicEOS-test.py]: this test only checks if the application doesn't crash, but I can add one regression here

File: Reaktoro/Thermodynamics/EOS/CubicEOS.cpp (hunk 2 above)
Comment (alexandrebbruno) [reply Reaktoro/Thermodynamics/EOS/CubicEOS.cpp]: done :)

File: Reaktoro/Thermodynamics/EOS/CubicEOS.cpp (hunk 3 above)
Comment (alexandrebbruno) [reply Reaktoro/Thermodynamics/EOS/CubicEOS.cpp]: ok, done

**Pull request merged**